gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java

```diff
@@ -0,0 +1,526 @@
+              record.ttl(), Joiner.on(", ").join(record.records()));
+        }
+      } else {
+        System.out.printf("Zone %s has no DNS records.%n", zoneName);
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name> records";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 2;
+    }
+  }
+
+  private static class ListChangesAction implements DnsAction {
+
+    /**
+     * Lists all the changes for a given zone. Optionally, an order, "descending" or "ascending" can
+     * be specified using the last parameter.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      Iterator<ChangeRequest> iterator;
+      if (args.length > 2) {
+        Dns.SortingOrder sortOrder = Dns.SortingOrder.valueOf(args[2].toUpperCase());
+        iterator = dns.listChangeRequests(zoneName,
+            Dns.ChangeRequestListOption.sortOrder(sortOrder)).iterateAll();
+      } else {
+        iterator = dns.listChangeRequests(zoneName).iterateAll();
+      }
+      if (iterator.hasNext()) {
+        System.out.printf("Change requests for zone %s:%n", zoneName);
+        while (iterator.hasNext()) {
+          ChangeRequest change = iterator.next();
+          System.out.printf("%nID: %s%n", change.id());
+          System.out.printf("Status: %s%n", change.status());
+          System.out.printf("Started: %s%n", FORMATTER.format(change.startTimeMillis()));
+          System.out.printf("Deletions: %s%n", Joiner.on(", ").join(change.deletions()));
+          System.out.printf("Additions: %s%n", Joiner.on(", ").join(change.additions()));
+        }
+      } else {
+        System.out.printf("Zone %s has no change requests.%n", zoneName);
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name> changes [descending | ascending]";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 2
+          || (args.length == 3
+          && ImmutableList.of("descending", "ascending").contains(args[2].toLowerCase()));
+    }
+  }
+
+  private static class ListAction implements DnsAction {
+
+    /**
+     * Invokes a list action. If no parameter is provided, lists all zones. If zone name is the only
+     * parameter provided, lists both DNS records and changes. Otherwise, invokes listing changes or
+     * zones based on the parameter provided.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      if (args.length == 0) {
+        new ListZonesAction().run(dns);
+      } else {
+        if (args.length == 1 || "records".equals(args[1])) {
+          new ListDnsRecordsAction().run(dns, args);
+        }
+        if (args.length == 1 || "changes".equals(args[1])) {
+          new ListChangesAction().run(dns, args);
+        }
+      }
+    }
+
+    @Override
+    public boolean check(String... args) {
+      if (args.length == 0) {
```

Comment:
I think I suggested removing the `|| args.length == 1` earlier, but I realize this would cause an NPE if the user provides only the zone name.  Let's add it back here.